crates/apub/src/objects/post.rs

```diff
@@ -179,12 +173,23 @@ impl ApubObject for ApubPost {
       } else {
         (None, thumbnail_url)
       };
-      let (embed_title, embed_description, embed_html) = metadata_res
+      let (mut embed_title, mut embed_description, mut embed_html) = metadata_res
         .map(|u| (u.title, u.description, u.html))
         .unwrap_or((None, None, None));
       let body_slurs_removed =
         read_from_string_or_source_opt(&page.content, &page.media_type, &page.source)
           .map(|s| remove_slurs(&s, &context.settings().slur_regex()));
+      // we cant display videos directly, so insert a link to external video page
+      if page.kind == PageType::Video {
+        url = Some(page.id.inner().clone());
+        // TODO: without these fields present, embed seems is not shown. needs to be fixed in lemmy-ui
+        embed_title = Some("test".to_string());
+        embed_description = Some("test".to_string());
+        // TODO: need to generate this in a better way. one option is parsing property "og:video:url"
+        //       of the peertube video page for embed url. but as this is federation code, i think
+        //       it would make sense to federate the embed code
+        embed_html = Some(r#"<iframe src="https://peertube2.cpy.re/videos/embed/0ee82ed7-2582-4851-856d-9b151c3e243c"></iframe>"#.to_string());
```

Comment:
We need to get the Peertube embed url properly. One option is to parse the site property "og:video:url", but then it will work for all websites, not only Peertube. And as this the federation code, i believe it would make sense to also send the embed url over activitypub. What do you think about this @chocobozzz?